web/js/containers/map-interactions/ol-vector-interactions.js

```diff
@@ -38,9 +46,81 @@ export class VectorInteractions extends React.Component {
     events.off('map:singleclick', this.singleClick);
   }
 
+  /**
+  * Handle mouse over granule geometry and trigger action to show granule date footprint
+  *
+  * @param {Object} granuleCMRGeometry
+  * @param {Object} map
+  * @param {String} crs
+  * @param {Array} pixels
+  * @param {Array} coord
+  *
+  * @return {Boolean} to indicate if valid for parent mouseMove function
+  */
+  handleGranuleHover = (granuleCMRGeometry, crs, pixels, coord) => {
+    const {
+      active,
+      activeString,
+      compareState,
+      granulePlatform,
+      maxExtent,
+      swipeOffset,
+    } = this.props;
+
+    let toggledGranuleFootprint;
+    // reverse granule geometry so most recent granules are on top
+    const gcmr = Object.keys(granuleCMRGeometry).reverse().map((key) => ({ [key]: granuleCMRGeometry[key] }));
+    // only allow hover footprints on selected side of A/B comparison
+    if (active) {
+      const isOnActiveCompareSide = isFromActiveCompareRegion(pixels, activeString, compareState, swipeOffset);
+      if (!isOnActiveCompareSide) {
+        events.trigger('granule-hovered', null);
```

Comment:
These events seem strange to me. If there is going to be action listeners outside of the redux approach, I feel like it should to be really documented/thought through since it kinda goes against the general structure of the app. Maybe we want to use React Context for when we are doing updates and we don't want to update the entire app.

Reply:
We have been using these in a number of places for a while to communicate things like mouse events across components.  I think it's a fairly standard and simple event pub/sub setup that's pretty easy to use though a little documentation couldn't hurt.  I haven't really used React Context and though I imagine it could allow us to do something similar, I'm not sure if it's intended to solve this kind of problem.  [The docs say](https://reactjs.org/docs/context.html): _"Context is primarily used when some data needs to be accessible by many components at different nesting levels. Apply it sparingly because it makes component reuse more difficult._

In our case, we don't need data accessible by many components, we just need to communicate between two places in the code (which may or may not even be React components) when an event has been triggered.

I did a bit of searching for pub/sub in React and [it seems like this is a common way](https://ralexanderson.com/blog/spreading-events-in-react-with-pubsub) to approach this kind of cross-component communication outside of the component tree.  I think for things like mouse hover events, this kind of event system is really the way to go.